src/NLSModels.jl

```diff
@@ -115,6 +117,25 @@ function jac_residual(nls :: AbstractNLSModel, x :: AbstractVector)
   throw(NotImplementedError("jac_residual"))
 end
 
+"""
+    (rows,cols) = jac_structure_residual(nls)
+
+Returns the structure of the constraint's Jacobian in sparse coordinate format.
+"""
+function jac_structure_residual(nls :: AbstractNLSModel)
+  throw(NotImplementedError("jac_structure_residual"))
+end
+
+"""
+    (rows,cols,vals) = jac_coord_residual!(nls, x, rows, cols, vals)
+
+Computes the Jacobian of the residual at `x` in sparse coordinate format, rewriting
+`vals`. `rows` and `cols` are not rewritten.
+"""
+function jac_coord_residual!(nls :: AbstractNLSModel, x :: AbstractVector, rows :: AbstractVector{Int}, cols :: AbstractVector{Int}, vals :: AbstractVector)
```

Comment:
Does this work? I thought `Vector{Int} <: AbstractVector{Int}` was `false`?!

Reply:
`Vector{Int} <: AbstractVector{Int}` is true, but `AbstractVector{Int32} <: AbstractVector{Int}` is false.